datafusion/sqllogictest/test_files/expr.slt

```diff
@@ -510,6 +510,42 @@ SELECT ltrim(NULL, 'xyz')
 ----
 NULL
 
+# implicit casting with TypeSignature test
+query T
+SELECT ltrim(NULL, NULL)
+----
+NULL
+
+query T
+SELECT ltrim(12345, '1')
```

Comment:
DuckDB doesn't allow this query 🤔 
Numeric string casting is quite confusing to me

```
D SELECT ltrim(12345, '1');
Binder Error: No function matches the given name and argument types 'ltrim(INTEGER_LITERAL, STRING_LITERAL)'. You might need to add explicit type casts.
	Candidate functions:
	ltrim(VARCHAR) -> VARCHAR
	ltrim(VARCHAR, VARCHAR) -> VARCHAR

LINE 1: SELECT ltrim(12345, '1');
```